R/req-url.R

```diff
@@ -10,7 +10,7 @@
 #' @param url New URL; completely replaces existing.
 #' @param ... For `req_url_query()`: <[`dynamic-dots`][rlang::dyn-dots]>
 #'   Name-value pairs that provide query parameters. Each value must be either
-#'   a length-1 atomic vector (which is automatically escaped) or `NULL` (which
+#'   an atomic vector (which is automatically escaped) or `NULL` (which
```

Comment:
That's actually not quite true that `NULL` is silently dropped. You can remove a query parameter with `NULL`:

``` r
request("https://example.com") |>
  req_url_query(foo = c(Sys.Date(), Sys.Date()), .multi = "explode") |>
  req_url_query(foo = NULL) |>
  req_dry_run()
#> GET / HTTP/1.1
#> Host: example.com
#> User-Agent: httr2/0.2.3.9000 r-curl/5.1.0 libcurl/8.1.2
#> Accept: */*
#> Accept-Encoding: deflate, gzip
```

<sup>Created on 2023-10-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>